index.js

```diff
@@ -98,6 +98,13 @@ CanvasRenderingContext2D.prototype.beginCircle = function (x, y, r) {
   this.arc(x, y, r, 0, 2.0 * Math.PI);
 };
 
+//create a String containing all the possible shortcodes for the shapes
+const possibleShapesString = Object.keys(enumShortcodeToSubShape).toString().replaceAll(',', '');
```

Comment:
`Object.keys()` always returns a normal array.
`Array.prototype.join()` is shorter, 50% faster, and easier to read.
```suggestion
const possibleShapesString = Object.keys(enumShortcodeToSubShape).join('');
```

Reply:
Line below also